tests/Unit/Controllers/UserControllerTest.php

```diff
@@ -0,0 +1,130 @@
+<?php
+
+namespace Tests\Unit\Controllers;
+
+use Mockery;
+use Tests\TestCase;
+use App\Http\Controllers\UserController;
+use App\Repositories\User\UserRepositoryInterface;
+use Illuminate\Database\Eloquent\Collection;
+use Illuminate\Database\Eloquent\ModelNotFoundException;
+use Illuminate\Http\RedirectResponse;
+use App\Http\Requests\RegisterRequest;
+use Illuminate\Http\Request;
+use App\Http\Requests\UpdateProfileRequest;
+use Faker\Factory as Faker;
+use Hash;
+
+
+class UserControllerTest extends TestCase
+{
+    protected $userController;
+    protected $userMock;
+    protected $faker;
+
+    protected function setUp(): void
+    {
+        $this->userMock = Mockery::mock(UserRepositoryInterface::class);
+        $this->userController = new UserController($this->userMock);
+        parent::setUp();
+    }
```

Comment:
em khởi tại gì ở đây?

Reply:
Em khởi tạo 1 Mock Object $this->userMock dựa trên UserRepositoryInterface
Sau đó, em khởi tạo 1 UserController có tham số truyền vào là mock object ở trên.